temporal-sdk/src/main/java/io/temporal/client/WorkflowOptions.java

```diff
@@ -151,9 +152,24 @@ public Builder setWorkflowIdReusePolicy(WorkflowIdReusePolicy workflowIdReusePol
     }
 
     /**
-     * The time after which workflow run is automatically terminated by Temporal service. Do not
-     * rely on run timeout for business level timeouts. It is preferred to use in workflow timers
-     * for this purpose.
+     * The time after which workflow run is automatically terminated by Temporal service with
+     * WORKFLOW_EXECUTION_TIMED_OUT status.
+     *
+     * <p>When workflow reaches Workflow Run Timeout, it can't make any process after that. Do not
+     * rely on this timeout in workflow implementation or business logic. Workflow run timeout is
+     * not designed to be handled in workflow code to perform any logic in case of timeout. Consider
+     * using workflow timers instead.
+     *
+     * <p>If you catch yourself setting this timeout to very small values, you likely use it wrong.
+     *
+     * <p>Example: If Workflow Run Timeout is 30 seconds and network was unavailable for 1 minute,
+     * workflows that got scheduled before the network blip will never get a chance to make progress
+     * and react and will be terminated. <br>
+     * A timer that is scheduled in the workflow code using {@link Workflow#newTimer(Duration)} will
+     * handle this situation gracefully. A workflow with such a timer will start after the network
+     * blip. If it started before the network blip and the timer fires during the network blip, it
+     * will get delivered after the connectivity is restored and the workflow will be able to
+     * resume.
```

Comment:
```suggestion
     * The time after which a workflow run is automatically terminated by Temporal service with
     * WORKFLOW_EXECUTION_TIMED_OUT status.
     *
     * <p>When a workflow reaches Workflow Run Timeout, it can't make any progress after that. Do not
     * rely on this timeout in workflow implementation or business logic. This timeout is
     * not designed to be handled in workflow code to perform any logic in case of timeout. Consider
     * using workflow timers instead.
     *
     * <p>If you catch yourself setting this timeout to very small values, you're likely using it wrong.
     *
     * <p>Example: If Workflow Run Timeout is 30 seconds and the network was unavailable for 1 minute,
     * workflows that were scheduled before the network blip will never have a chance to make progress
     * or react, and will be terminated. <br>
     * A timer that is scheduled in the workflow code using {@link Workflow#newTimer(Duration)} will
     * handle this situation gracefully. A workflow with such a timer will start after the network
     * blip. If it started before the network blip and the timer fires during the network blip, it
     * will get delivered after connectivity is restored and the workflow will be able to
     * resume.
```